jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py

```diff
@@ -796,31 +830,57 @@ def calculate_photometry(self, dataset=None, aperture=None, background=None,
 
             phot_table.add_column(slice_val, name="slice_wave", index=29)
 
-            if comp.units:  # convert phot. results from image unit to display unit
+            if comp.units:
+
+                # convert units of output table to reflect display units
+                # selected in Unit Conversion plugin
                 display_unit = u.Unit(self.display_unit)
-                # convert units of certain columns in aperture phot. output table
-                # to reflect display units (i.e if data units are MJy / sr, but
-                # Jy / sr is selected in Unit Conversion plugin)
-                if display_unit != '':
-                    phot_table['background'] = phot_table['background'].to(
-                        display_unit, u.spectral_density(self._cube_wave))
 
+                # equivalencies for unit conversion, will never be flux<>sb
+                # so only need spectral_density
+                equivs = u.spectral_density(self._cube_wave)
+
+                if display_unit != '':
+                    if phot_table['background'].unit != display_unit:
+                        bg_conv = flux_conversion_general(phot_table['background'].value,
+                                                          phot_table['background'].unit,
+                                                          display_unit,
+                                                          equivs)
+                        phot_table['background'] = bg_conv
+
+                    phot_sum = phot_table['sum']
                     if include_pixarea_fac:
-                        phot_table['sum'] = phot_table['sum'].to(
-                            (display_unit * pixarea_fac).unit, u.spectral_density(self._cube_wave))
+                        if phot_sum.unit != display_unit * pixarea_fac:
+                            phot_table['sum'] = flux_conversion_general(phot_sum.value,
+                                                                        phot_sum.unit,
+                                                                        (display_unit * pixarea_fac).unit,  # noqa

```

Comment:
Should be more explicit on which flake8 warning we are ignoring. That said, this could be written such that ignoring is not necessary in the first place but I understand everyone's style is different.
```suggestion
                                                                        (display_unit * pixarea_fac).unit,  # noqa: E501
```